packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2300,12 +2300,50 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     }
   }
 
+  Set<SemanticsAction> _mutedSemanticsActions = <SemanticsAction>{};
+
+  /// Update the set of [SemanticsAction] that will be ignore when constructing
+  /// semantics action handlers.
+  ///
+  /// The muted [SemanticsAction] will apply to all the children in subtree.
+  void updateMutedSemanticsActions(Set<SemanticsAction> actions) {
+    assert(actions != null);
+    if (_mutedSemanticsActions.length == actions.length &&
+        _mutedSemanticsActions.every(actions.contains))
+      return;
+    _mutedSemanticsActions = actions;
+    markNeedsSemanticsUpdate();
+    bool markRenderObjectNeedsSemanticsUpdate(RenderObject renderChild) {
```

Comment:
I believe you do not need to walk down to the leaves. If you mark a node as needing semantics update, the entire subtree gets re-build anyways starting with the marked node.

Reply:
its subtree already cached both _cachedMutedSemanticsActions and _cachedSemanticsConfiguration which will need to be removed by calling markNeedsSemanticsUpdate. I feel a bit scared to manually remove the cache without going through markNeedsSemanticsUpdate.